compiler/src/dotty/tools/dotc/core/Names.scala

```diff
@@ -163,7 +163,7 @@ object Names {
     def info = NameInfo.TermName
     def underlying: TermName = unsupported("underlying")
 
-    private var derivedNames: AnyRef /* SimpleMap | j.u.HashMap */ =
+    @sharable private var derivedNames: AnyRef /* SimpleMap | j.u.HashMap */ =
```

Comment:
Why is this safe? `synchronized` is not used when modifying `derivedNames` currently.